### ISSUE TITLE
making the mapper function as optional parameters

### ISSUE THREAD

**Issue by ZibanPirate** [ZibanPirate/l2t#7]
**Is your feature request related to a problem? Please describe.**
Currently the last parameter (mapper function) is mandatory, again this function is great for even more flexibility, but sometimes the case is simple, and you don't want to map the object, but instead return it as is.
```ts
// convert the simple list into a tree
const tree = listToTree(
  // the list
  simpleList,
  // a function returns id of the item
  (item) => item.id,
  // a function returns id of the item's parent
  (item) => item.parentId,
  // key for storing children items if there are any
  "children",
  // a mapper function to map the item to whatever object you like
  (item) => {
    return {
      label: `item-number-${item.label}`, // let's modify the label and prepend "item-number-"
      index: item.id, // let's change id to index
    };
  },
);
```
**Describe the solution you'd like**
a solution would be a non-breaking change, buy making the last parameter optional
```ts
// convert the simple list into a tree
const tree = listToTree(
  // the list
  simpleList,
  // a function returns id of the item
  (item) => item.id,
  // a function returns id of the item's parent
  (item) => item.parentId,
  // key for storing children items if there are any
  "children",
);
```

**Comment by sainitishkumar** [comment]
Hi, can I take this up?

**Comment by ZibanPirate** [comment]
Hey @sainitishkumar, yes please do 🔥

**Comment by sainitishkumar** [comment]
Hey, I have changed the code for making the mapper function as an optional parameter.

But when I test the original code with `yarn test`, it gives an error on line 27 and 28 in src/index.ts

Can you please check that?

**Comment by ZibanPirate** [comment]
please create an MR so that i can get more context, you can use the `--no-verify` flag to ignore test check, and will fix test after

**Comment by sainitishkumar** [comment]
Can you check this out? #8

**Comment by ZibanPirate** [comment]
@sainitishkumar seems all good, tests are passing on my machine (MacOS) as well as CI pipeline (Ubuntu), you only need to update the tests to cover the optional case
please run `yarn test --coverage` to check the coverage percentage, it has to be 100%

**Comment by sainitishkumar** [comment]
Hi, can you help me test the code?

**Comment by ZibanPirate** [comment]
sure, make sure to first create a non-mapped result object, you can add it to the top of test file (after `simpleTree` object)
```js
const simpleTreeNotMapped = [
  {
    id: "1",
    label: "1",
    children: [
      { id: "2", label: "2", parentId: "1", children: [] },
      { id: "3", label: "3", parentId: "1", children: [] },
    ],
  },
  { id: "4", label: "4", children: [] },
  {
    id: "5",
    label: "5",
    children: [{ id: "6", label: "6", parentId: "5", children: [] }],
  },
];
```

and then you test against that object, so you modify your test to look like this:
```js
test("Simple List to Tree without Mapper function", () => {
  const tree = listToTree(
    simpleList,
    (item) => item.id,
    (item) => item.parentId,
    "children",
  );

  expect(tree).toMatchObject(simpleTreeNotMapped);
});
```

please make sure to do the same for for `es6.spec` and `typescript.spec` as well

**Comment by sainitishkumar** [comment]
Hi! Thanks for the help with tests.
Please review this #8 PR

**Comment by ZibanPirate** [comment]
closing as it's fixed in #8 thanks to @sainitishkumar
